Refactor `ext/pgsql` test

KentarouTakeda opened the pull request:
I refactored the `ext/pgsql` tests based on some recent pull requests.
The content and results remain unchanged. This is just an internal improvement.

With this refactoring, we can expect the following benefits:

* Test independence:
  Some tests were dependent on the execution order of previous and subsequent tests.
  All tests can now be executed independently.
* Test parallelism
  One test no longer interferes with another. All tests can be run in parallel.

Specifically, i made the following modifications: For readability, each modification is fixed in a separate commit.

* Use `--CLEAN--` to delete objects at the end of the test.
  https://github.com/php/php-src/commit/faabc3fba80f5b098373fb18cc3347c84b9f34e2
* Moved `*.inc` to another folder to make filenames easier to read.
  https://github.com/php/php-src/commit/7b507cd072babec6d0d75a88c31b418be4b9517c
* Tables used for tests are defined for each test instead of being set all at once in `config.inc`
  https://github.com/php/php-src/commit/002fe321e0bd10fbb31ebb6fcd0b1015ff1fb6e7 https://github.com/php/php-src/commit/293b0b01eacef73a7281dd988ffc82046e9148d6
* Use different object names for each test. The name is based on the test file name.
  https://github.com/php/php-src/commit/002fe321e0bd10fbb31ebb6fcd0b1015ff1fb6e7 https://github.com/php/php-src/commit/5ab45c70b9c5829493ece0f0b5cba522e4764e3d

Ref:
* https://github.com/php/php-src/pull/11855
* https://github.com/php/php-src/pull/11879
* https://github.com/php/php-src/pull/12555

Girgias commented on ext/pgsql/tests/80_bug24499.phpt:
```diff
@@ -48,6 +46,13 @@ pg_close($dbh);
+require_once('inc/config.inc');
+$dbh = pg_connect($conn_str);
+
+pg_query($dbh, "DROP TABLE id CASCADE");
```
Probably should be `DROP TABLE IF EXISTS`

KentarouTakeda replied:
Fixed in https://github.com/php/php-src/pull/12608/commits/8a3484888dce8530ada1c4c698a1923f180020ca
All existing code unrelated to this pull request was also fixed.

Girgias commented on ext/pgsql/tests/01createdb.phpt:
```diff
@@ -9,14 +9,14 @@ pgsql
 {
-    pg_query($db,$table_def); // Create table here
-    for ($i=0; $i < $num_test_record; $i++) {
-        pg_query($db,"INSERT INTO ".$table_name." VALUES ($i, 'ABC');");
-    }
+    pg_query($db, "CREATE TABLE {$table_name} (num int, str text, bin bytea)"); // Create table here
```
This is not inserting data any more?

KentarouTakeda replied:
OK. This is no longer needed.

This test is originally a test for `cratedb` as the test file name suggests.
However, before the fix in this pull request was made, it was also responsible for inserting data used in subsequent tests. I fixed it as follows.

> * Test independence:
> Some tests were dependent on the execution order of previous and subsequent tests.
> All tests can now be executed independently.

In other words, instead of inserting it as a "common process" at this point, it has been modified to insert the data required for each test individually.

Girgias replied:
ACK

Girgias commented on ext/pgsql/tests/bug37100_9.phpt:
```diff
@@ -41,9 +42,10 @@ pg_close($db);
 <?php
 require_once('inc/config.inc');
-$db = pg_connect($conn_str);
+$table_name = 'ttable_bug37100_9';
```
``'table_bug37100_9'``

KentarouTakeda replied:
Fixed in https://github.com/php/php-src/pull/12608/commits/6918e6ff2af5caf8b2c01dfb2264a885399fbf34

Girgias reviewed (changes requested):
Thank you for working on this!

It has been very annoying to need to run these tests sequentially.

Could you always use uppercase letters for SQL keywords, and CLEAN sections should use an IF EXISTS clause as the table may not be created in the test.

KentarouTakeda commented:
> Could you always use uppercase letters for SQL keywords,

Fixed in https://github.com/php/php-src/pull/12608/commits/21e61ad1f21a8360297aa8ba659d4306ee0a6ed7

All existing code unrelated to this pull request was also fixed.
However, there were some inconsistencies in the interpretation of what should be treated as "SQL keywords". For those, I respect the original code and have not fixed it.

Girgias commented on ext/pgsql/tests/01createdb.phpt:
```diff
@@ -25,18 +25,28 @@ else {
+
+$db = pg_connect($conn_str);
+pg_query($db, "DROP TABLE IF EXISTS {$table_name} cascade");
+pg_query($db, "DROP TABLE if exists {$table_name_92} cascade");
```
```suggestion
pg_query($db, "DROP TABLE IF EXISTS {$table_name_92} cascade");
```

KentarouTakeda replied:
Thanks!!

Girgias commented on ext/pgsql/tests/9999dropdb.phpt:
```diff
@@ -3,15 +3,22 @@ PostgreSQL drop db
+pg_query($db, "CREATE VIEW {$view_name} as SELECT * FROM {$table_name}");
+
+pg_query($db, "DROP VIEW IF EXISTS {$view_name}");
 pg_query($db, "DROP TABLE ".$table_name);
 @pg_query($db, "DROP TABLE ".$table_name_92);
```
```suggestion
pg_query($db, "DROP TABLE IF EXISTS " . $table_name);
pg_query($db, "DROP TABLE IF EXISTS " . $table_name_92);
```
Or maybe use the ``{$var}`` interpolation syntax?

KentarouTakeda replied:
Fixed in https://github.com/php/php-src/pull/12608/commits/89aa3f76c30f2444d2f856511d7ddda584fbdba7

Sorry, besides this, I overlooked many other deficiencies in `IF EXISTS`. Modified to include `{$var}` .

Girgias commented on ext/pgsql/tests/bug47199.phpt:
```diff
@@ -31,11 +30,18 @@ echo $query, "\n";
+$dbh = pg_connect($conn_str);
+
+$tbl_name = 'test_47199';
+pg_query($dbh, "DROP TABLE $tbl_name");
```
```suggestion
pg_query($dbh, "DROP TABLE IF EXISTS $tbl_name");
```

KentarouTakeda replied:
Thanks!!

Girgias commented on ext/pgsql/tests/bug68638.phpt:
```diff
@@ -30,8 +30,15 @@ while ($row = pg_fetch_assoc($rs)) {
+$conn = pg_connect($conn_str);
+$table='test_68638';
+
+pg_query($conn, "DROP TABLE $table");
```
```suggestion
pg_query($conn, "DROP TABLE IF EXISTS $table");
```

KentarouTakeda replied:
Thanks!!

Girgias commented on ext/pgsql/tests/bug71062.phpt:
```diff
@@ -30,10 +30,16 @@ pg_convert($db, $table, ['test_field' => $date_string_modified_iso8601]);
+$db = @pg_connect($conn_str);
+$table = "public.test_table_bug71062_bug71062";
+
+pg_query($db, "DROP TABLE $table");
```
```suggestion
pg_query($db, "DROP TABLE IF EXISTS $table");
```

KentarouTakeda replied:
Thanks!!

Girgias commented on ext/pgsql/tests/bug72028.phpt:
```diff
@@ -33,9 +33,14 @@ $r = pg_query($conn, "SELECT * FROM $table");
+$table = "bug72028";;
+$conn = pg_connect($conn_str);
 
 pg_query($conn, "DROP TABLE $table");
```
```suggestion
pg_query($conn, "DROP TABLE IF EXISTS $table");
```

KentarouTakeda replied:
Thanks!!

Girgias commented on ext/pgsql/tests/pg_select_001.phpt:
```diff
@@ -3,49 +3,55 @@ PostgreSQL pg_select() - basic test using schema
+
+$conn = pg_connect($conn_str);
 
+pg_query($conn, "DROP SCHEMA IF EXISTS {$schema_name} CASCADE");
```
Shouldn't we be dropping the 2 tables created in the test?

KentarouTakeda replied:
Fixed in https://github.com/php/php-src/pull/12608/commits/c6ec16c7193cc1be42fa5cc7cf96f342527fdb26

In fact, the behavior will not change even if we do not modify it.
The `cascade` option will also drop tables that depend on the schema being dropped.

However, based on your point, I thought it would be easier to understand if I explicitly deleted it for the sake of clarity. Thank you!

Girgias reviewed (commented):
Looks mostly good now! Only a couple of nits and questions. :)

KentarouTakeda commented:
@Girgias 
Your very detailed review helped me a lot!
I have corrected each of the points pointed out. thank you.

Girgias commented on ext/pgsql/tests/9999dropdb.phpt:
```diff
@@ -3,17 +3,24 @@ PostgreSQL drop db
+pg_query($db, "CREATE TABLE {$table_name_92} (textary text[],  jsn json)");
+pg_query($db, "CREATE VIEW {$view_name} as SELECT * FROM {$table_name}");
+
+pg_query($db, "DROP VIEW IF EXISTS {$view_name}");
+pg_query($db, "DROP TABLE IF EXISTS {$table_name}");
+pg_query($db, "DROP TABLE IF EXISTS {$table_name_92}");
 
 echo "OK";
```
This should be in a CLEAN section

KentarouTakeda replied:
Fixed in https://github.com/php/php-src/pull/12608/commits/eadb74cb36cae1a0326cc984f176de95630033c1

Just to be safe, I dropped it in both the test and `--CLEAN--`, and removed `IF EXISTS` in the test.
Since the test name suggests that it is a `DROP` test, I thought that if it fails, it should be reported as a test failure.

Girgias reviewed (approved):
One final remark and good to merge :)

Girgias commented:
Thank you!
